src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -198,12 +198,33 @@ public function setRequired(bool $boolean)
     /**
      * Sets this node as deprecated.
      *
+     * @param string $package The name of the composer package that is triggering the deprecation
+     * @param string $version The version of the package that introduced the deprecation
+     * @param string $message The deprecation message to use
+     *
      * You can use %node% and %path% placeholders in your message to display,
      * respectively, the node name and its complete path.
      */
-    public function setDeprecated(?string $message)
+    public function setDeprecated(/* string $package, string $version, string $message */)
     {
-        $this->deprecationMessage = $message;
+        $args = \func_get_args();
+
+        if (func_num_args() < 3) {
+            trigger_deprecation('symfony/config', '5.1', 'The signature of "%s()" method requires 3 arguments: "string $package, string $version, string $message", not defining them is deprecated.', __METHOD__);
```

Comment:
we also need a separate deprecation when passing `null` as argument, as the call now need to be avoided (we cannot un-deprecate a node anymore or make a no-op call to avoid marking as deprecated)

Reply:
I added a separate deprecation and a unit test for this case